Issue #18: Use the Heartbeat v2 API.

Castaglia opened the pull request:
## Pull Request Checklist

fixes #18 

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### Purpose
EDIT: Use v2 apis.
#### Known Compatability Issues
EDIT: drop ruby 2.0 support and testing because it is EOL

Commit:
Issue #18: Use the Heartbeat v2 API.

Commit:
Update the CHANGELOG for the fix.

majormoses commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+- Use Heartbeat v2 API
```
This is a needs to go under a section header such as `### Changed`, `### Added`, `### Breaking Changes`, `### Fixed` as outlined [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md). Can you verify that the api key provided for v1 works on v2 or do they have different auth mechanisms. I am just asking because I have seen that kind of stuff before and I am not really familiar with `opsgenie`

Castaglia replied:
I not able to verify whether the same API key can be used for v1 and v2; I only have access to the v2 APIs.  Attempting to use my current API key for the v1 Heartbeat API results in an HTTP 401 response.

majormoses replied:
@eheydrick can you check on this since I belive you use this?

Castaglia replied:
Updated PR to list this change under a "Fixed" section.

majormoses commented on bin/check-opsgenie-heartbeat.rb:
```diff
@@ -74,17 +75,12 @@ def run
-
-    uri = URI.parse('https://api.opsgenie.com/v1/json/heartbeat/send')
+    encoded_name = URI.escape(config[:name])
+    uri = URI.parse("https://api.opsgenie.com/v2/heartbeats/#{encoded_name}/ping")
```
Do we need the api key here as well?

Castaglia replied:
The API key, in the v2 API, is provided via the `Authorization` header; it is not part of the URI, nor is it a query parameter.

majormoses replied:
makes sense.

majormoses commented:
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

majormoses reviewed (changes requested):
Overall looks good, some minor tweaks, answer a few questions, and provide a [testing artifact](https://github.com/sensu-plugins/community/blob/master/PULL_REQUEST_PROCESS.md#7-testing-artifacts) are needed.

Commit:
Update the ChangeLog, per review feedback.

Castaglia commented:
There are no current/existing tests for this particular plugin, to demonstrate what such preferable test artifacts might be; could you point me in the direction of what you have in mind?

majormoses commented:
Sure, so the link I put before for [testing artifacts](https://github.com/sensu-plugins/community/blob/master/PULL_REQUEST_PROCESS.md#7-testing-artifacts) talk about testing in the PR processes. I'd either go for a manual test or some integration tests with `serverspec` and `test-kitchen`. I have wrote a pair of blog posts talking more about plugin testing:
- https://blog.sensuapp.org/our-approach-to-testing-sensu-plugins-869260eba94c
- https://blog.sensuapp.org/writing-sensu-plugin-tests-with-test-kitchen-and-serverspec-b646d2eeee51

Commit:
Use kitchen-vagrant to create some basic tests.

majormoses committed:
update some of the testing stuff:
- standard `Rakefile`
- standard `.kitchen.yml`
- switch from `vagrant` to `docker` as it is lighter weight and speeds up testing time.
- test multiple rubies
- setup gem deps

Signed-off-by: Ben Abrams <me@benabrams.it>

Castaglia commented:
Thanks for the fixes, @majormoses .  My Ruby-fu is still pretty weak. :)

majormoses commented:
No problem I am more than happy to help anyone willing to put in a little effort. I will get the testing framework functional for you.

majormoses commented:
If it makes you feel any better I am sure your c skills are quite a bit better than mine. I have not written anything in c for a while.

majormoses commented:
@eheydrick any comments as a user of this? If I don't hear back shortly I will merge this.

eheydrick commented:
The code looks good to me.

majormoses commented:
released: https://rubygems.org/gems/sensu-plugins-opsgenie/versions/4.0.0
